### PR TITLE
Better gatewayd logs

### PR DESCRIPTION
**Pull request by Maan2003**
fixes https://github.com/fedimint/fedimint/issues/4165
based on #4174

### CODE REVIEW

**Review by elsirion — APPROVED** [review]

File: fedimint-core/src/util/mod.rs [hunk 1]
```diff
@@ -217,6 +218,81 @@ pub async fn write_new_async<P: AsRef<Path>, C: AsRef<[u8]>>(
         .await
 }
 
+#[derive(Debug, Clone)]
+pub struct Spanned<T> {
+    value: T,
+    span: Span,
+}
```
[the remark this answers is not in the document]
Comment (Maan2003) [reply fedimint-core/src/util/mod.rs:225]: didn't find anything on quick web search

[the remark this answers is not in the document]
Comment (elsirion) [reply fedimint-core/src/util/mod.rs:225]: Might be worth upstreaming.

**Review by m1sterc001guy — APPROVED** [review]
LGTM

**Comment by fedimint-backports** [comment]
Backport failed for `releases/v0.2`, because it was unable to cherry-pick the commit(s).

Please cherry-pick the changes locally and resolve any conflicts.
```bash
git fetch origin releases/v0.2
git worktree add -d .worktree/backport-4175-to-releases/v0.2 origin/releases/v0.2
cd .worktree/backport-4175-to-releases/v0.2
git switch --create backport-4175-to-releases/v0.2
git cherry-pick -x 3ce9a5a6a380c03d613335190b5f7af7f2945d33 d4858d43141be2b8a7425ab53d22214bb72c416a 24c3eef191b75a9f7b8b053c7dff8a71a769894f
```

**Comment by Maan2003** [comment]
looks like backports port is just bad now

**Comment by elsirion** [comment]
We probably just diverged too much already, still good for opening issues.

**Pull request merged**